### PR TITLE
Implement Boilerplate for Generic Nodes

### PR DESCRIPTION
**Pull request by noanflaherty**
I'll soon be writing unit tests for our `graph` attribute. As part of this, I'm gonna want to keep my tests very basic and just work with custom/generic nodes, rather than instantiating specific nodes of specific types.

As such, I figured it's as good a time as any to introduce the boilerplate needed for generic nodes.

### CODE REVIEW

File: ee/codegen/src/serializers/vellum.ts [hunk 1]
```diff
@@ -1655,6 +1656,29 @@ export declare namespace ErrorNodeSerializer {
...
+> = objectSchema({
+  type: stringLiteralSchema("GENERIC"),
+  id: stringSchema(),
+  data: objectSchema({
+    label: stringSchema(),
+  }),
+  inputs: listSchema(NodeInputSerializer),
+  displayData: propertySchema("display_data", anySchema()),
+  definition: WorkflowNodeDefinitionSerializer.optional(),
+});
```
Comment (noanflaherty) [inline ee/codegen/src/serializers/vellum.ts]: I expect this to change a lot down the line, but starting with this for now.

**Review by NgoHarrison — APPROVED** [review]

**Review by peicodes — APPROVED** [review]

File: ee/codegen/src/__test__/helpers/node-data-factories.ts [hunk 2]
```diff
@@ -872,3 +873,15 @@ export function ApiNodeFactory({
...
 }
+
+export function genericNodeFactory(): GenericNode {
+  const nodeData: GenericNode = {
+    id: "7e09927b-6d6f-4829-92c9-54e66bdcaf80",
+    type: WorkflowNodeType.GENERIC,
+    data: {
+      label: "Generic Node",
+    },
+    inputs: [],
```
Comment (dvargasfuertes) [inline ee/codegen/src/__test__/helpers/node-data-factories.ts]: [q] do we need `id`, `data` and `inputs`?

**Review by dvargas92495 — APPROVED** [review]
Ideally we are able to only include fields that we perceive existing long term. If the goal is "to play nicely with the UI today", we'd be able to make a hard fork on the logic based on `type=='GENERIC'`, and go down a component tree that doesn't assume old fields. Once we migrate all old nodes to be generic nodes, we could then drop the `type`

**Comment by noanflaherty** [comment]
I'm going to close this and reopen following Vargas's feedback. There's like other smaller PRs I can get in to ease us into this.

**Comment by noanflaherty** [comment]
@dvargas92495 this is ready for re-review.

cc @peicodes

**Review by dvargas92495 — APPROVED** [review]
💥

**Pull request merged**